conformance/utils/kubernetes/helpers.go

```diff
@@ -271,3 +307,42 @@ func GetGatewayEndpoint(t *testing.T, k8sClient client.Client, timeoutConfig gat
 	t.Logf("Gateway %s/%s has address: %s", gatewayNN.Namespace, gatewayNN.Name, gwAddr)
 	return gwAddr
 }
+
+// GetPod waits for a Pod matching the specified labels to exist in the given
+// namespace and have an IP address assigned. This function returns the first
+// matching Pod found if there are multiple matches. It fails the on timeout or error.
+func GetPod(t *testing.T, c client.Client, namespace string, selector labels.Selector, timeout time.Duration) *corev1.Pod {
```

Comment:
I think @zetxqx is adding (or maybe already has added) a util that allows getting a list of Pods like this. Recommend converging on a single approach for this.